sway-lib-std/src/storage/storage_vec.sw

```diff
@@ -755,10 +783,131 @@ impl<V> StorageKey<StorageVec<V>> {
+    // let other_vec = Vec::from(storage.storage_vec);
+    // ```
+    /// Stores a `Vec` as a `StorageVec`.
+    ///
+    /// # Additional Information
+    ///
+    /// This will overwrite any existing values in the `StorageVec`.
+    ///
+    /// # Arguments
+    /// 
+    /// * `vec`: [Vec<V>] - The vector to store in storage.
+    ///
+    /// # Number of Storage Accesses
+    ///
+    /// * Writes - `2`
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     let mut vec = Vec::<u64>::new();
+    ///     vec.push(5);
+    ///     vec.push(10);
+    ///     vec.push(15);
+    ///
+    ///     storage.vec.store_vec(vec);
+    ///
+    ///     assert(5 == storage.vec.get(0).unwrap());
+    ///     assert(10 == storage.vec.get(1).unwrap());
+    ///     assert(15 == storage.vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(write)]
+    pub fn store_vec(self, vec: Vec<V>) {
+        let slice = vec.as_raw_slice();
+        // Get the number of storage slots needed based on the size of bytes.
+        let number_of_bytes = slice.number_of_bytes();
+        let number_of_slots = (number_of_bytes + 31) >> 5;
+        let mut ptr = slice.ptr();
+
+        // The capacity needs to be a multiple of 32 bytes so we can 
+        // make the 'quad' storage instruction store without accessing unallocated heap memory.
+        ptr = realloc_bytes(ptr, number_of_bytes, number_of_slots * 32);
+
+        // Store `number_of_slots * 32` bytes starting at storage slot `key`.
+        let _ = __state_store_quad(sha256(self.field_id), ptr, number_of_slots);
+
+        // Store the length, NOT the bytes. 
+        // This differs from the existing `write_slice()` function to be compatible with `StorageVec`.
+        write::<u64>(self.field_id, 0, number_of_bytes / __size_of::<V>());
+    }
+
+    /// Load a `Vec` from the `StorageVec`.
+    ///
+    /// # Returns
+    /// 
+    /// * [Option<Vec<V>>] - The vector constructed from storage or `None`.
+    ///
+    /// # Number of Storage Accesses
+    ///
+    /// * Reads - `2`
+    ///
+    /// # Examples
+    ///
+    /// ```sway
+    /// storage {
+    ///     vec: StorageVec<u64> = StorageVec {},
+    /// }
+    ///
+    /// fn foo() {
+    ///     let mut vec = Vec::<u64>::new();
+    ///     vec.push(5);
+    ///     vec.push(10);
+    ///     vec.push(15);
+    ///
+    ///     storage.vec.store_vec(vec);
+    ///     let returned_vec = storage.vec.load_vec().unwrap();
+    ///
+    ///     assert(5 == returned_vec.get(0).unwrap());
+    ///     assert(10 == returned_vec.get(1).unwrap());
+    ///     assert(15 == returned_vec.get(2).unwrap());
+    /// }
+    /// ```
+    #[storage(read)]
+    pub fn load_vec(self) -> Option<Vec<V>> {
+        // Get the length of the slice that is stored.
+        match read::<u64>(self.field_id, 0).unwrap_or(0) {
+            0 => None,
```

Comment:
I think you should just match on the Option here, I would expect an empty vec to be returned if there was an empty storagevec (I believe you can store an empty vec using the above store method?).